rasa/core/featurizers.py

```diff
@@ -645,21 +723,32 @@ def training_states_and_actions(
                         )
                     idx += 1
 
-        logger.debug("Created {} action examples.".format(len(trackers_as_actions)))
+            logger.debug("Created {} action examples.".format(len(trackers_as_actions)))
 
         return trackers_as_states, trackers_as_actions
 
     def prediction_states(
-        self, trackers: List[DialogueStateTracker], domain: Domain
+        self, trackers: List[DialogueStateTracker], domain: Domain, interpreter
     ) -> List[List[Dict[Text, float]]]:
         """Transforms list of trackers to lists of states for prediction."""
+        trackers_as_states = [self._create_states_e2e(tracker) for tracker in trackers]
+        
+        trackers_as_states_modified = []
+        for tracker in trackers_as_states:
+            curr_tracker = []
+            for state in tracker:
+                curr_state ={}
+                for key, value in state.items():
+                    if isinstance(value, Message):
+                        curr_state[key] = interpreter.parse(value.text)                    
```

Comment:
What `interpreter.parse` returns?

Reply:
essentially Message().as_dict()

Reply:
why do we need the `if` above? actually why do we modify `trackers_as_states`?